src/async_fuse/memfs/cache/backend.rs

```diff
@@ -0,0 +1,491 @@
+//! The implementation of backend.
+
+use async_trait::async_trait;
+use clippy_utilities::OverflowArithmetic;
+use datenlord::config::{StorageParams, StorageS3Config};
+use futures::{stream, AsyncReadExt, AsyncWriteExt, StreamExt};
+use opendal::services::{Fs, S3};
+use opendal::{ErrorKind, Operator};
+use tracing::warn;
+
+use super::{Block, Storage};
+use crate::async_fuse::fuse::protocol::INum;
+
+/// A builder to build `BackendWrapper`.
+#[derive(Debug)]
+pub struct BackendBuilder {
+    /// The storage config
+    config: StorageParams,
+    /// The size of a block
+    block_size: usize,
+}
+
+impl BackendBuilder {
+    /// Create a backend builder.
+    #[must_use]
+    pub fn new(config: StorageParams, block_size: usize) -> Self {
+        Self { config, block_size }
+    }
+
+    /// Build the backend.
+    pub fn build(self) -> opendal::Result<BackendWrapper> {
+        let BackendBuilder { config, block_size } = self;
```

Comment:
Is there a valid range for `block_size`? eg : [4k,4mb]  If so, use `debug_assert!` to enforce it

Reply:
No, there is no a valid range. The only limitation is that all storage layers must be with the same `block_size`, which will be guarnteed on creating the `StorageManager`. And in the future, we will record `block_size` in metadata, and this field here will be removed.